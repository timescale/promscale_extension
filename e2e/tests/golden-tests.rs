use crate::common::{exec_sql_script, run_postgres};
use insta::assert_snapshot;
use std::env;
use test_generator::test_resources;
use testcontainers::clients;

mod common;

#[test_resources("testdata/*.sql")]
fn golden_test(resource: &str) {
    let docker = clients::Cli::default();
    let node = run_postgres(&docker);
    let query_result = exec_sql_script(&node, resource);

    assert_snapshot!(resource, query_result);
}
