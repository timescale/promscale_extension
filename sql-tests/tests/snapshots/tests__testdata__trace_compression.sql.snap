---
source: e2e/tests/golden-tests.rs
expression: query_result
---
\set ON_ERROR_STOP 1
CREATE EXTENSION promscale;
CREATE EXTENSION
-- We don't want retention to mess with the test data
SELECT ps_trace.set_trace_retention_period('100 years'::INTERVAL);
 set_trace_retention_period 
----------------------------
 t
(1 row)

INSERT INTO _ps_trace.span(trace_id,span_id,parent_span_id,operation_id,start_time,end_time,duration_ms,trace_state,span_tags,dropped_tags_count,event_time,dropped_events_count,dropped_link_count,status_code,status_message,instrumentation_lib_id,resource_tags,resource_dropped_tags_count,resource_schema_url_id)
    VALUES
        (E'18dd078e-8c69-e10a-d2fe-9e9f47de7728',-2771219554170079234,NULL,19,E'2022-04-26 11:44:55.185139+00',E'2022-04-26 11:44:55.38517+00',200.031,NULL,E'{"1003": 247}',0,E'["2022-04-26 11:44:55.185659+00","2022-04-26 11:44:55.385148+00")',0,0,E'error',E'Exception: FAILED to fetch a lower char',5,E'{"1": 114, "5": 94, "6": 93, "7": 95}',0,NULL),
        (E'15a8be0f-bb79-c052-223e-48608580efce',2625299614982951051,NULL,19,E'2022-04-26 11:44:55.185962+00',E'2022-04-26 11:44:55.288812+00',102.85,NULL,E'{"1003": 242}',0,E'["2022-04-26 11:44:55.185999+00","2022-04-26 11:44:55.288781+00")',0,0,E'error',E'Exception: FAILED to fetch a lower char',5,E'{"1": 114, "5": 94, "6": 93, "7": 95}',0,NULL);
INSERT 0 2
SELECT hypertable_schema, hypertable_name,  chunk_schema, chunk_name, is_compressed FROM timescaledb_information.chunks order by range_end desc;
 hypertable_schema | hypertable_name |     chunk_schema      |    chunk_name    | is_compressed 
-------------------+-----------------+-----------------------+------------------+---------------
 _ps_trace         | span            | _timescaledb_internal | _hyper_7_1_chunk | f
(1 row)

CALL _ps_trace.execute_tracing_compression_job(0, jsonb '{"log_verbose":false,"hypertable_name":"span"}'); --noop
CALL
SELECT hypertable_schema, hypertable_name,  chunk_schema, chunk_name, is_compressed FROM timescaledb_information.chunks order by range_end desc;
 hypertable_schema | hypertable_name |     chunk_schema      |    chunk_name    | is_compressed 
-------------------+-----------------+-----------------------+------------------+---------------
 _ps_trace         | span            | _timescaledb_internal | _hyper_7_1_chunk | f
(1 row)

INSERT INTO _ps_trace.span(trace_id,span_id,parent_span_id,operation_id,start_time,end_time,duration_ms,trace_state,span_tags,dropped_tags_count,event_time,dropped_events_count,dropped_link_count,status_code,status_message,instrumentation_lib_id,resource_tags,resource_dropped_tags_count,resource_schema_url_id)
    VALUES
        (E'28dd078e-8c69-e10a-d2fe-9e9f47de7728',-2771219554170079234,NULL,19,E'2022-04-26 12:44:55.185139+00',E'2022-04-26 11:44:55.38517+00',200.031,NULL,E'{"1003": 247}',0,E'["2022-04-26 11:44:55.185659+00","2022-04-26 11:44:55.385148+00")',0,0,E'error',E'Exception: FAILED to fetch a lower char',5,E'{"1": 114, "5": 94, "6": 93, "7": 95}',0,NULL),
        (E'25a8be0f-bb79-c052-223e-48608580efce',2625299614982951051,NULL,19,E'2022-04-26 12:44:55.185962+00',E'2022-04-26 11:44:55.288812+00',102.85,NULL,E'{"1003": 242}',0,E'["2022-04-26 11:44:55.185999+00","2022-04-26 11:44:55.288781+00")',0,0,E'error',E'Exception: FAILED to fetch a lower char',5,E'{"1": 114, "5": 94, "6": 93, "7": 95}',0,NULL);
INSERT 0 2
CALL _ps_trace.execute_tracing_compression_job(0, jsonb '{"log_verbose":false,"hypertable_name":"span"}'); --compress one
CALL
SELECT hypertable_schema, hypertable_name,  chunk_schema, chunk_name, is_compressed FROM timescaledb_information.chunks order by range_end desc;
 hypertable_schema | hypertable_name |     chunk_schema      |    chunk_name    | is_compressed 
-------------------+-----------------+-----------------------+------------------+---------------
 _ps_trace         | span            | _timescaledb_internal | _hyper_7_2_chunk | f
 _ps_trace         | span            | _timescaledb_internal | _hyper_7_1_chunk | t
(2 rows)

CALL _ps_trace.execute_tracing_compression_job(0, jsonb '{"log_verbose":false,"hypertable_name":"span"}'); --noop
CALL
SELECT hypertable_schema, hypertable_name,  chunk_schema, chunk_name, is_compressed FROM timescaledb_information.chunks order by range_end desc;
 hypertable_schema | hypertable_name |     chunk_schema      |    chunk_name    | is_compressed 
-------------------+-----------------+-----------------------+------------------+---------------
 _ps_trace         | span            | _timescaledb_internal | _hyper_7_2_chunk | f
 _ps_trace         | span            | _timescaledb_internal | _hyper_7_1_chunk | t
(2 rows)

INSERT INTO _ps_trace.span(trace_id,span_id,parent_span_id,operation_id,start_time,end_time,duration_ms,trace_state,span_tags,dropped_tags_count,event_time,dropped_events_count,dropped_link_count,status_code,status_message,instrumentation_lib_id,resource_tags,resource_dropped_tags_count,resource_schema_url_id)
    VALUES
        (E'38dd078e-8c69-e10a-d2fe-9e9f47de7728',-2771219554170079234,NULL,19,E'2022-04-26 13:44:55.185139+00',E'2022-04-26 11:44:55.38517+00',200.031,NULL,E'{"1003": 247}',0,E'["2022-04-26 11:44:55.185659+00","2022-04-26 11:44:55.385148+00")',0,0,E'error',E'Exception: FAILED to fetch a lower char',5,E'{"1": 114, "5": 94, "6": 93, "7": 95}',0,NULL),
        (E'35a8be0f-bb79-c052-223e-48608580efce',2625299614982951051,NULL,19,E'2022-04-26 13:44:55.185962+00',E'2022-04-26 11:44:55.288812+00',102.85,NULL,E'{"1003": 242}',0,E'["2022-04-26 11:44:55.185999+00","2022-04-26 11:44:55.288781+00")',0,0,E'error',E'Exception: FAILED to fetch a lower char',5,E'{"1": 114, "5": 94, "6": 93, "7": 95}',0,NULL);
INSERT 0 2
SELECT hypertable_schema, hypertable_name,  chunk_schema, chunk_name, is_compressed FROM timescaledb_information.chunks order by range_end desc;
 hypertable_schema | hypertable_name |     chunk_schema      |    chunk_name    | is_compressed 
-------------------+-----------------+-----------------------+------------------+---------------
 _ps_trace         | span            | _timescaledb_internal | _hyper_7_4_chunk | f
 _ps_trace         | span            | _timescaledb_internal | _hyper_7_2_chunk | f
 _ps_trace         | span            | _timescaledb_internal | _hyper_7_1_chunk | t
(3 rows)

CALL _ps_trace.execute_tracing_compression_job(0, jsonb '{"log_verbose":false,"hypertable_name":"span"}'); --compress another one
CALL
SELECT hypertable_schema, hypertable_name,  chunk_schema, chunk_name, is_compressed FROM timescaledb_information.chunks order by range_end desc;
 hypertable_schema | hypertable_name |     chunk_schema      |    chunk_name    | is_compressed 
-------------------+-----------------+-----------------------+------------------+---------------
 _ps_trace         | span            | _timescaledb_internal | _hyper_7_4_chunk | f
 _ps_trace         | span            | _timescaledb_internal | _hyper_7_2_chunk | t
 _ps_trace         | span            | _timescaledb_internal | _hyper_7_1_chunk | t
(3 rows)

CALL _ps_trace.execute_tracing_compression_job(0, jsonb '{"log_verbose":false,"hypertable_name":"span"}'); --noop
CALL
SELECT hypertable_schema, hypertable_name,  chunk_schema, chunk_name, is_compressed FROM timescaledb_information.chunks order by range_end desc;
 hypertable_schema | hypertable_name |     chunk_schema      |    chunk_name    | is_compressed 
-------------------+-----------------+-----------------------+------------------+---------------
 _ps_trace         | span            | _timescaledb_internal | _hyper_7_4_chunk | f
 _ps_trace         | span            | _timescaledb_internal | _hyper_7_2_chunk | t
 _ps_trace         | span            | _timescaledb_internal | _hyper_7_1_chunk | t
(3 rows)


